src/track/beatgrid.cpp

```diff
@@ -184,13 +190,49 @@ double BeatGrid::findNthBeat(double dSamples, int n) const {
         n = n + 1;
     }
 
-    double dResult = dClosestBeat + n * m_dBeatLength;
+    double dResult = floor(dClosestBeat + n * m_dBeatLength);
     if (!even(static_cast<int>(dResult))) {
         dResult--;
     }
     return dResult;
```

Comment:
Eh, let's not prematurely optimize here. The root issue is that we're locking a mutex at some fixed rate across threads. I don't believe CPU time in these methods has been demonstrated as an issue so saving one or two *instructions* here or there isn't helpful.